compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/viewinterop/InteropViewHolder.skiko.kt

```diff
@@ -18,31 +18,192 @@ package androidx.compose.ui.viewinterop
+    throw NotImplementedError("Abstract `$name` must be implemented by platform-specific subclass of `InteropViewHolder`")
+}
 
 /**
  * A holder that keeps references to user interop view and its group (container).
- * It's actual implementation of [InteropViewFactoryHolder]
+ * It's an actual implementation of `expect class [InteropViewFactoryHolder]`
  *
  * @see InteropViewFactoryHolder
+ *
+ * @param platformModifier The modifier that is specific to the platform.
  */
 internal open class InteropViewHolder(
     val container: InteropContainer,
-    val group: InteropViewGroup
+    val group: InteropViewGroup,
+    private val compositeKeyHash: Int,
+    measurePolicy: MeasurePolicy,
+    isInteractive: Boolean,
+    platformModifier: Modifier
 ) : ComposeNodeLifecycleCallback {
+    private var onModifierChanged: ((Modifier) -> Unit)? = null
+
+    /**
+     * User-provided modifier that will be reapplied if changed.
+     */
+    var modifier: Modifier = Modifier
+        set(value) {
+            if (value !== field) {
+                field = value
+                onModifierChanged?.invoke(value)
+            }
+        }
+
+    private var hasUpdateBlock = false
+
+    var update: () -> Unit = {}
+        protected set(value) {
+            field = value
+            hasUpdateBlock = true
+            runUpdate()
+        }
+
+    protected var reset: () -> Unit = {}
+
+    protected var release: () -> Unit = {}
+
+    private var onDensityChanged: ((Density) -> Unit)? = null
+    var density: Density = Density(1f)
+        set(value) {
+            if (value !== field) {
+                field = value
+                onDensityChanged?.invoke(value)
+            }
+        }
+
+    /**
+     * If the view is not attached, update on closure change (or on setting initial one) will
+     * be postponed until it's attached and triggered when this flag is set to `true`.
+     *
+     * If the view is detached, the observer is stopped to avoid redundant callbacks.
+     */
+    private var isAttachedToWindow: Boolean = false
+        set(value) {
+            if (value != field) {
+                field = value
+
+                if (value) {
+                    runUpdate()
+                } else {
+                    snapshotObserver.clear(this)
+                }
+            }
+        }
+
+    private val snapshotObserver: SnapshotStateObserver
+        get() {
+            return container.snapshotObserver
+        }
+
+    private val runUpdate: () -> Unit = {
+        // If we're not attached, the observer isn't started, so don't bother running it.
+        // onAttachedToWindow will run an update the next time the view is attached.
+        if (hasUpdateBlock && isAttachedToWindow) {
+            snapshotObserver.observeReads(this, OnCommitAffectingUpdate, update)
+        }
+    }
+
+    override fun onReuse() {
+        reset()
+    }
+
+    override fun onDeactivate() {
+    }
```

Comment:
Which seems redundant in our case. 
Because documentations states this:
```
    /**
     * Invoked when the group containing the node was deactivated.
     * This happens when the content of [ReusableContentHost] is deactivated.
     *
     * The node will not be reused in this recompose cycle, but might be reused or released in
     * the future. Consumers might use this callback to release expensive resources or stop
     * continuous process that was dependent on the node being used in composition.
     *
     * If the node is reused, [onReuse] will be called again to prepare the node for reuse.
     * Similarly, [onRelease] will indicate that deactivated node will never be reused again.
     */
    fun onDeactivate()
```

It means `reset` will be called for a reused view?

Reply:
> Which seems redundant in our case

Aren't we going to use this as common (including Android) in the future?

Reply:
Then we shall discuss with Google, why Android does this, because it doesn't make sense

Reply:
Will leave a reminder here